app-modules/ai/src/Filament/Resources/PromptResource/Pages/EditPrompt.php

```diff
@@ -87,6 +91,32 @@ public function form(Form $form): Form
                             ])
                             ->grouped()
                             ->visible(auth()->user()->hasRole(Authenticatable::SUPER_ADMIN_ROLE)),
+                        Group::make()->schema([
+                            Checkbox::make('is_confidential')
+                                ->label('Confidential')
+                                ->live()
+                                ->columnSpanFull(),
+                            Select::make('confidential_prompt_users')
+                                ->relationship('confidentialAccessUsers', 'name')
+                                ->preload()
+                                ->label('Users')
+                                ->multiple()
+                                ->exists('users', 'id')
+                                ->visible(fn (Get $get) => $get('is_confidential')),
+                            Select::make('confidential_prompt_teams')
+                                ->relationship('confidentialAccessTeams', 'name')
+                                ->preload()
+                                ->label('Teams')
+                                ->multiple()
+                                ->exists('teams', 'id')
+                                ->visible(fn (Get $get) => $get('is_confidential')),
+                        ])
+                            ->columns(2)
+                            ->columnSpanFull()
+                            ->visible(fn (Get $get) => ConfidentialPromptsFeature::active()
+                                && auth()->user()->hasRole(Authenticatable::SUPER_ADMIN_ROLE)
+                                || auth()->user()->id == $get('user_id')
+                                && $get('is_smart') == false),
```

Comment:
Please make a habit of using strict comparison. It is only niche cases where loose comparison is necessary imo.

```suggestion
                                || auth()->user()->id === $get('user_id')
                                && $get('is_smart') === false),
```